filebeat/input/kafka/kafka_integration_test.go

```diff
@@ -0,0 +1,265 @@
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package kafka
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	_ "github.com/elastic/beats/libbeat/outputs/codec/format"
+	_ "github.com/elastic/beats/libbeat/outputs/codec/json"
+)
+
+const (
+	kafkaDefaultHost = "kafka"
+	kafkaDefaultPort = "9092"
+)
+
+type eventInfo struct {
+	events []beat.Event
+}
+
+type eventCapturer struct {
+	closed    bool
+	c         chan struct{}
+	closeOnce sync.Once
+	events    chan beat.Event
+}
+
+func NewEventCapturer(events chan beat.Event) channel.Outleter {
+	return &eventCapturer{
+		c:      make(chan struct{}),
+		events: events,
+	}
+}
+
+func (o *eventCapturer) OnEvent(event beat.Event) bool {
+	o.events <- event
+	return true
+}
+
+func (o *eventCapturer) Close() error {
+	o.closeOnce.Do(func() {
+		o.closed = true
+		close(o.c)
+	})
+	return nil
+}
+
+func (o *eventCapturer) Done() <-chan struct{} {
+	return o.c
+}
+
+type testMessage struct {
+	message string
+	headers []sarama.RecordHeader
+}
+
+func recordHeader(key, value string) sarama.RecordHeader {
+	return sarama.RecordHeader{
+		Key:   []byte(key),
+		Value: []byte(value),
+	}
+}
+
+func TestInput(t *testing.T) {
+	id := strconv.Itoa(rand.New(rand.NewSource(int64(time.Now().Nanosecond()))).Int())
+	testTopic := fmt.Sprintf("Filebeat-TestInput-%s", id)
+	context := input.Context{
+		Done:     make(chan struct{}),
+		BeatDone: make(chan struct{}),
+	}
+
+	// Send test messages to the topic for the input to read.
+	messages := []testMessage{
+		testMessage{message: "testing"},
+		testMessage{
+			message: "stuff",
+			headers: []sarama.RecordHeader{
+				recordHeader("X-Test-Header", "test header value"),
+			},
+		},
+		testMessage{
+			message: "things",
+			headers: []sarama.RecordHeader{
+				recordHeader("keys and things", "3^3 = 27"),
+				recordHeader("kafka yay", "3^3 - 2^4 = 11"),
+			},
+		},
+	}
+	for _, m := range messages {
+		writeToKafkaTopic(t, testTopic, m.message, m.headers, time.Second*20)
+	}
+
+	// Setup the input config
+	config, _ := common.NewConfigFrom(common.MapStr{
```

Comment:
Use `MustNewConfigFrom`. This would panic, potentially pointing out programmer error when setting up the test.